oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Utils.java

```diff
@@ -1,130 +1,92 @@
+import com.google.common.base.Strings;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.jackrabbit.core.data.DataStoreException;
 import org.apache.jackrabbit.oak.commons.PropertiesUtil;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-public final class Utils {
-
-    public static final String DEFAULT_CONFIG_FILE = "azure.properties";
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Properties;
 
+public class Utils {
     public static final String DASH = "-";
+    public static final String DEFAULT_CONFIG_FILE = "azure.properties";
 
-    /**
-     * private constructor so that class cannot initialized from outside.
-     */
-    private Utils() {
-    }
-
-    /**
-     * Create CloudBlobClient from properties.
-     *
-     * @param connectionString connectionString to configure @link {@link CloudBlobClient}
-     * @return {@link CloudBlobClient}
-     */
-    public static CloudBlobClient getBlobClient(@NotNull final String connectionString) throws URISyntaxException, InvalidKeyException {
-        return getBlobClient(connectionString, null);
-    }
+    public Utils() {}
 
-    public static CloudBlobClient getBlobClient(@NotNull final String connectionString,
-                                                @Nullable final BlobRequestOptions requestOptions) throws URISyntaxException, InvalidKeyException {
-        CloudStorageAccount account = CloudStorageAccount.parse(connectionString);
-        CloudBlobClient client = account.createCloudBlobClient();
-        if (null != requestOptions) {
-            client.setDefaultRequestOptions(requestOptions);
-        }
-        return client;
-    }
+    public static BlobContainerClient getBlobContainer(String accountName, String accountKey, @NotNull final String connectionString,
+                                                       @NotNull final String containerName,
+                                                       @Nullable final RequestRetryOptions retryOptions,
+                                                       boolean isProxyNeeded, final Properties properties) throws DataStoreException {
+        BlobContainerClient blobContainerClient;
+        try {
+            if(isProxyNeeded) {
+                String proxyHost = properties.getProperty(AzureConstants.PROXY_HOST);
+                String proxyPort = properties.getProperty(AzureConstants.PROXY_PORT);
+
+                ProxyOptions proxyOptions = new ProxyOptions(ProxyOptions.Type.HTTP,
+                        new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort)));
+
+                HttpClient httpClient = new NettyAsyncHttpClientBuilder()
+                        .proxy(proxyOptions)
+                        .build();
+
+                BlobServiceClient blobServiceClient = new BlobServiceClientBuilder()
```

Comment:
lines 49-55 are duplicates with 58-63. the method should be refactored and do apply some abstractisation.